src/TypeSpec.Extension/Emitter.Csharp/src/lib/model.ts

```diff
@@ -579,8 +579,14 @@ export function getInputType(
             const discriminatorProperty = m.properties.get(
                 discriminatorPropertyName
             );
-            if (discriminatorProperty?.type.kind === "String") {
-                return discriminatorProperty.type.value;
+            if (
```

Comment:
It would be good to have unit test for each change we made in Emitter. 
But as this is just one line change, I will leave it for you to decide.

Reply:
I cannot add a test case, because right now the compiler will directly report error.